### ISSUE TITLE
[JIT] Add bool type to the schema (fix torch.var)

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#8758]
`torch.var` and `torch.std` in jit script exhibit incorrect behavior.

```
import torch
@torch.jit.script
def fn(x):
    dim = 1
    return torch.std(x, dim)

x = torch.randn(2, 3)
fn(x)  # gives  tensor(0.8668)
torch.std(x, 1)  # gives tensor([1.2670, 0.4717])
```

This is because of the following:

there are two schemas for var (and std):
(1) var(Tensor, int, *, bool, bool, Tensor)
(2) var(Tensor, *, bool)

Right now, the following is happening:
- Shorter schemas come before longer schemas
- bool, int are treated as IntType rather than DynamicType like before
So the schemas look like the following in aten_schema:
(2) var(DynamicType, IntType)
(1) var(DynamicType, IntType, IntType, DynamicType)

Now, when one calls torch.var(tensor, dim=1), the compiler mistakingly matches it with (2) instead of (1), which is a problem.

**Comment by driazati** [comment]
Fixed in #10921
